src/smc-webapp/project_files.cjsx

```diff
@@ -1994,14 +1994,24 @@ exports.ProjectFiles = rclass ({name}) ->
                 style   = {error_style}
                 onClose = {=>@props.actions.setState(error:'')} />
 
-    render_not_public_error: ->
-        if @props.redux.getStore('account').is_logged_in()
-            <ErrorDisplay title="Directory is not public" error={"You are trying to access a non public project that you are not a collaborator on. You need to ask a collaborator of the project to add you."} />
+    render_not_collaborator_error: ->
+        public_view = @props.get_my_group(@props.project_id) == 'public'
+        if public_view
+            if @props.redux.getStore('account').is_logged_in()
+                <ErrorDisplay title="Showing only public files" error={"You are trying to access a project that you are not a collaborator on. To view non-public files or edit files in this project you need to ask a collaborator of the project to add you."} />
+            else
+                <div>
+                    <ErrorDisplay title="Showing only public files" error={"You are not logged in. To view non-public files or edit files in this project you need to ask a collaborator of the project to add you."} />
```

Comment:
hmm, I'm probably confused, but this case here should include an emphasis on having to log in, just like the else case below (i.e. "If you are collaborator on this project you need to log in first.") -- or am I missing something?